content/english/secret-messages/activity-5.md

```diff
@@ -0,0 +1,80 @@
+
+Example
+```
+char letter = 'A';
+int letter_value = letter;
+
+cout << letter << " = " << letter_value << endl;
+```
+This will print:
+```
+A = 65
+```
+
+In this example, we save the character in a variable type **char**. Then, in an **int** type variable we save the char type variabale, but since the second variable saves numbers, it will convert the character into its numerical value according to the ascii code.A
+
+In this way, we could do the opposite, for example:
+```
+int letter_value = 67;
+char letter = letter_value;
+cout << letter_value << " = " << letter << endl;
+```
+This will print:
+```
+67 = C
+```
+{{% /notice %}}
+
+To create the secret message we will use a keyword, from which we will encode the name. The idea is to go through each character of the name with a **for** statement, and at the same time go through each character of the keyword. Then we will add the value of the character of the name and the keyword, and the result will be a new character, which will make the message encrypted.
+
+For example, if my keyword is `queen` and the name I want to encrypt is `programmer`, the program has to add the `'r'` and the `'p'`, and save the result in place of 'p' in the string "programmer". Then I would add the  `'r'` and the `'e'`, the  `'o'` and the `'i'` , etc. When we are in character 5 of the programmer string (counted the first as 0), we will have to turn around in the queen string and start in the returned character 0, that is, the letter `'r'`, since the queen string has no more characters.
+
+For that, we are going to gather part of what we have already done, and we must follow the following steps:
+
+1. Creates a variable of type string to be the keyword.
+2. Enter its value per console.
+3. Declares a variable type int and initializes it to 0. To this variable, we will add 1 in each cycle of the for to go through the keyword.
+4. In the for statement, create an int variable that is equal to the sum of the name character (`'name[i]'`) and the keyword character at the counter position (`'key[counter]'`). From this sum we must subtract a lowercase letter `'a'` so that the sum corresponds to a letter and not another character of the ascii code.
+5. Uses an if statement to determine whether the sum of characters corresponds to a character greater than `'z'`. If so, this number must be subtracted from 25 (the number of letters of the alphabet) so that the number of the turn, and begin with the letter 'a'.
+6. Equalizes the character `'name[i]'` to the variable int with the sum. This will change the original letter to the new character.
+7. Add 1 to the counter we carry from the keyword.
+8. Schedule a condition where if the keyword counter is greater than or equal to the length of the key string, the counter will be worth 0 again.
+9. Finally, after the for statement, print the various name by console.
```

Comment:
```suggestion
1. First, create a variable of type `string` to be the keyword.
2. Store the input received from console into this variable. This will be the name.
3. Repeat the first two steps, but this will store the keyword.
3. Declare a `for` loop, and use a variable of type `int` initialized to 0. Add `1` to this variable for each cycle of the loop, and continue the loop while the variable is less than the `string` variable's `size`.
4. In the body of the `for` loop, create an `int` variable that is equal to the sum of the name character (`name[i]`) and the keyword character at the counter position (`key[counter]`). From this sum we must subtract a lowercase letter `'a'` so that the sum corresponds to a letter and not some other random character of the ASCII code.
5. Use an `if` statement to determine whether the sum of characters corresponds to a character greater than `'z'`. If so, subtract by `25` (the number of letters of the alphabet) to ensure we get a true letter in the alphabet.
6. Set the character `name[i]` to be the value contained by the `sum` variable. This will change the original letter to the new character.
7. Add `1` to the counter we carry from the keyword.
8. Check if the keyword counter is greater than or equal to the length of the `key` string; if so, reset the counter to `0`.
9. Finally, after the `for` statement, print the encrypted message to the console.
```